development/gcp/cloud-run/kms-rotate/kms-rotate.go

```diff
@@ -0,0 +1,234 @@
+// package main contains code for KMS GCP symmetric key secret re-encrypting
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+
+	kms "cloud.google.com/go/kms/apiv1"
+	"cloud.google.com/go/storage"
+	"github.com/go-playground/validator/v10"
+	"google.golang.org/api/iterator"
+	kmspb "google.golang.org/genproto/googleapis/cloud/kms/v1"
+)
+
+type versionCountError struct {
+	Msg string
+}
+
+func (vce versionCountError) Error() string {
+	return vce.Msg
+}
+
+// Config contains function configuration provided through POST JSON
+type Config struct {
+	Project      string `json:"project" validate:"required,min=1"`
+	Location     string `json:"location" validate:"required,min=1"`
+	BucketName   string `json:"bucketName" validate:"required,min=1"`
+	BucketPrefix string `json:"bucketPrefix,omitempty"`
+	Keyring      string `json:"keyring" validate:"required,min=1"`
+	Key          string `json:"key" validate:"required,min=1"`
+}
+
+var (
+	kmsService     *kms.KeyManagementClient
+	storageService *storage.Client
+)
+
+func main() {
+	var err error
+	ctx := context.Background()
+
+	kmsService, err = kms.NewKeyManagementClient(ctx)
+	if err != nil {
+		log.Fatal("failed creating KMS client, error: " + err.Error())
+	}
+	defer kmsService.Close()
```

Comment:
shouldn't we return here?

Reply:
log.Fatal() prints out a message and calls `os.Exit()`